epsagon/events/pymongo.py

```diff
@@ -70,17 +74,81 @@ def update_response(self, response):
-            self.resource['metadata']['items'][i]['_id'] = str(
-                self.resource['metadata']['items'][i]['_id']
-            )
 
-        if self.resource['operation'] == 'insert_many':
-            self.resource['metadata']['inserted_ids'] = \
-                [str(x) for x in response.inserted_ids]
-        elif self.resource['operation'] == 'insert_one':
+        if self.resource['operation'] in PyMongoEvent.INSERT_OPERATIONS:
+            self.handle_insert_operations_response(response)
+
+        elif self.resource['operation'] in PyMongoEvent.FILTER_OPERATIONS:
+            self.handle_filter_operations_response(response)
+
+
+    def handle_request_payload(self, input_args):
+        """
+        Handle input data before add to event.
+        :param input_args: input from botocore
+        :return: None
+        """
+
+        if self.resource['operation'] == PyMongoEvent.INSERT_MANY:
+            add_data_if_needed(self.resource['metadata'], 'Items', input_args)
+
+        elif self.resource['operation'] == PyMongoEvent.INSERT_ONE:
+            add_data_if_needed(self.resource['metadata'], 'Item', input_args)
+
+        elif self.resource['operation'] in PyMongoEvent.FILTER_OPERATIONS:
+            add_data_if_needed(self.resource['metadata'], 'Filter', input_args)
+
+
+    def handle_insert_operations_response(self, response):
+        """
+        Handle response data before add to event.
+        :param response: response from botocore
+        :return: None
+        """
+
+        if self.resource['operation'] == PyMongoEvent.INSERT_MANY:
+            for i in range(len(self.resource['metadata'].get('items', []))):
+                self.resource['metadata']['Items'][i]['_id'] = str(
+                    self.resource['metadata']['Items'][i]['_id']
+                )
+
             self.resource['metadata']['inserted_ids'] = \
-                [str(response.inserted_id)]
+                    [str(x) for x in response.inserted_ids]
+
+        elif self.resource['operation'] == PyMongoEvent.INSERT_ONE:
+            for i in range(len(self.resource['metadata'].get('Item', []))):
+                self.resource['metadata']['Item'][i]['_id'] = str(
+                    self.resource['metadata']['Item'][i]['_id']
+                )
+
+            self.resource['metadata']['inserted_id'] = \
+                    [str(response.inserted_id)]
+
+
+    def handle_filter_operations_response(self, response):
+        """
+        Handle response data before add to event.
+        :param response: response from botocore
+        :return: None
+        """
+
+        self.resource['metadata']['Filter'] = str(
+                    self.resource['metadata']['Filter']
+                )
+
+        if self.resource['operation'] == 'find':
+            self.resource['metadata']['Results'] = \
+                        list(response)
+
+        elif self.resource['operation'] in ['update_one']:
```

Comment:
better check if its equal to update_one (if we will have one more similar operation, we can add the "in" check)